docs/libraries/typespec-client-generator-core/reference/linter.md

```diff
@@ -0,0 +1,31 @@
+---
+title: "Linter usage"
+toc_min_heading_level: 2
+toc_max_heading_level: 3
+---
+
+# Linter
+
+## Usage
+
+Add the following in `tspconfig.yaml`:
+
+```yaml
+linter:
+  extends:
+    - "@azure-tools/typespec-client-generator-core/all"
+```
+
+## RuleSets
+
+Available ruleSets:
+
+- `@azure-tools/typespec-client-generator-core/all`
```

Comment:
I am not sure `all` would ever be useful when considering non tier 1 languages as well as required rules vs best practices?  Should we be a bit more granular on what `all` means?

Reply:
`all` isn't a specificly-defined ruleset, it's the standard name for all of the rules existing in a package